src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Globalization;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class GenerateCurrentVersion : Task
+    {
+        /// <summary>
+        /// The passed in date that will be used to generate a version. (yyyy-MM-dd format)
+        /// </summary>
+        [Required]
+        public string SeedDate { get; set; }
+
+        /// <summary>
+        /// Optional parameter that sets the Padding for the version number.
+        /// </summary>
+        public string Padding { get; set; }
+
+        /// <summary>
+        /// If basing off of internal builds version format is not required, this optional parameter lets you pass in a comparison date.
+        /// </summary>
+        public string ComparisonDate { get; set; }
+
+        /// <summary>
+        /// The Major Version that will be produced given a SeedDate.
+        /// </summary>
+        [Output]
+        public string GeneratedVersion { get; set; }
+
+        private const string DateFormat = "yyyy-MM-dd";
+        private CultureInfo enUS = new CultureInfo("en-US");
+        private int padding;
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrEmpty(Padding))
+            {
+                Padding = "5";
+            }
+            bool isValidPadding = int.TryParse(Padding, out padding);
```

Comment:
You can actually just make the property an int because msbuild supports that type.